test/metabase/driver/mysql_test.clj

```diff
@@ -283,33 +286,44 @@
 
 (deftest group-on-time-column-test
-  (let [driver :mysql]
-    (mt/test-driver driver
-      (let [db-name "time_test"
-            spec (sql-jdbc.conn/connection-details->spec :mysql (tx/dbdef->connection-details driver :server nil))]
-        (doseq [stmt ["DROP DATABASE IF EXISTS time_test;"
-                      "CREATE DATABASE time_test;"]]
-          (jdbc/execute! spec [stmt]))
-        (let [details (tx/dbdef->connection-details driver :db {:database-name db-name})
-              spec (sql-jdbc.conn/connection-details->spec driver details)]
-          (doseq [stmt ["DROP TABLE IF EXISTS time_table;"
-                        "CREATE TABLE time_table (id serial, mytime time);"
-                        "INSERT INTO time_table (mytime) VALUES ('00:00'), ('00:00'), ('23:01'), ('23:01'), ('18:43');"]]
-            (jdbc/execute! spec [stmt]))
-          (mt/with-temp Database [db {:engine driver :details details}]
-            (sync/sync-database! db)
-            (mt/with-db db
-              (testing "can group on TIME columns"
-                (let [now (ZonedDateTime/now (ZoneId/of "UTC"))
-                      now-date-str (.format now (DateTimeFormatter/ISO_LOCAL_DATE))
-                      add-date-fn (fn [t] [(str now-date-str "T" t)])]
-                  (is (= (map add-date-fn ["00:00:00Z" "18:43:00Z" "23:01:00Z"])
-                         (mt/rows
-                           (mt/run-mbql-query time_table
-                             {:breakout [[:datetime-field (mt/id :time_table :mytime) :minute]]
-                              :order-by [[:asc [:datetime-field (mt/id :time_table :mytime) :minute]]]}))))
-                  (is (= (map add-date-fn ["23:00:00Z" "18:00:00Z" "00:00:00Z"])
-                         (mt/rows
-                           (mt/run-mbql-query time_table
-                             {:breakout [[:datetime-field (mt/id :time_table :mytime) :hour]]
-                              :order-by [[:desc [:datetime-field (mt/id :time_table :mytime) :hour]]]})))))))))))))
+  (mt/test-driver :mysql
+    (testing "can group on TIME columns (#12846)"
+      (mt/dataset attempted-murders
+        (let [now-date-str (u.date/format (t/local-date (t/zone-id "UTC")))
+              add-date-fn  (fn [t] [(str now-date-str "T" t)])]
+          (testing "by minute"
+            (is (= (map add-date-fn ["00:14:00Z" "00:23:00Z" "00:35:00Z"])
+                   (mt/rows
+                     (mt/run-mbql-query attempts
+                       {:breakout [!minute.time]
+                        :order-by [[:asc !minute.time]]
+                        :limit    3})))))
+          (testing "by hour"
+            (is (= (map add-date-fn ["23:00:00Z" "20:00:00Z" "19:00:00Z"])
+                   (mt/rows
+                     (mt/run-mbql-query attempts
+                       {:breakout [!hour.time]
+                        :order-by [[:desc !hour.time]]
+                        :limit    3}))))))))))
```

Comment:
not specifically related to this fix, but I rewrote this test to use the existing `attempted-murders` dataset instead of creating a new one, which is faster and a lot simpler